openmc/material.py

```diff
@@ -41,11 +41,19 @@ class Material(object):
     name : str, optional
         Name of the material. If not specified, the name will be the empty
         string.
+    temperature : str, optional
+        The temperature identifier applied to this material. The units are
+        in Kelvin and the temperature rounded to the nearest integer.
+        For example, a tempreature of 293.6K would be provided as '294K'
 
     Attributes
     ----------
     id : int
         Unique identifier for the material
+    temperature : str
```

Comment:
The more I think abou this the more OK with it I become.  On the one hand putting `temperature` in `Cell` is both more correct and more useful when using temperature feedback, but on the other, when you are not using temperature feedback or depletion the inputs become way more verbose as there are way more cells than materials in your standard BOC, zero power case.

Given we have the pyAPI now (and probably should keep a default_temperature somewhere), verbosity isnt such a big deal anymore, so I guess I agree, lets go with correctness and put temperature only in the `Cell`.

Further, thinking about what to do on the code side, you'd need to do have the following steps included in the input processing:
 1. Read geometry.xml and materials.xml
 2. For every nuclide/sab identify the temperatures you need by parsing the geometry
 3. If interpolating on temperature, for each nuclide:
 
   a. Identify the upper and lower bounds of all temperatures needed, if we are really advanced you could also identify multiple sets of boundaries (as a simple example, O-16 is in the water and the fuel. In the fuel the range of temperatures is in >1000K whereas in the water the range of temperatures is much closer to 600K.  You'd like to say 'i dont need 700K-1000K' if possible).
    b. load all data at all temperature sets within your temperature ranges needed

 4. If using WMP or other future Doppler broadening algorithm, for each nuclide:

    a. Same as 3.a, identify your ranges
    b. load only the data you need at all temperature sets within your temperature ranges needed (i.e., reaction/product/distribution data, URR data vs T, and alll sab data)
    c. load the WMP data

 5. Otherwise we are just loading the nearest data sets to the temperatures from the geometry for each nuclide